content/src/service/snapshots/SnapshotManager.ts

```diff
@@ -140,76 +159,90 @@ export class SnapshotManager {
   }
 
   /** This methods queries the database and builds the snapshots, stores it on the content storage, and saves the metadata */
-  private async generateSnapshot(task?: Database): Promise<void> {
+  private async generateSnapshot(): Promise<void> {
+    // Format the snapshot in a tmp file
+    const tmpFile = path.resolve(this.contentStorageFolder, 'tmp-snapshot-file')
+
+    const { end: stopTimer } = metricsComponent.startTimer('dcl_content_snapshot_generation_time')
+
     try {
+      if (await checkFileExists(tmpFile)) {
```

Comment:
why do we need this? Can we add a comment here?

Reply:
```suggestion
      // if the process failed while creating the snapshot last time the file may still exists
      // deleting the staging tmpFile just in case
      if (await checkFileExists(tmpFile)) {
```